v2/audit/conn/tcp.go

```
@@ -0,0 +1,41 @@
+//go:build linux || darwin
+
+package conn
+
+import (
+	"net"
+
+	"github.com/microsoft/go-otel-audit/v2/audit/conn/internal/writer"
+)
+
+// Compile check on interface implementation.
+var _ Audit = TCPConn{}
+
+// TCPConn represents a connection to a remote audit server via a TCP socket
+// This implements conn.Audit interface.
+type TCPConn struct {
+	*writer.Conn
+}
+
+// Type returns the type of the audit connection.
+func (TCPConn) Type() Type {
+	return TypeTCP
+}
+
+func (TCPConn) private() {}
+
+// NewTCPConn creates a new connection to the remote audit server. addr is the host:port of the remote audit server.
+// host can be an IP address or a hostname. If host is a hostname, it will be resolved to an IP address.
+func NewTCPConn(addr string) (TCPConn, error) {
+	conn, err := net.Dial("tcp", addr)
+	if err != nil {
+		return TCPConn{}, err
+	}
+	w, err := writer.New(conn)
+	if err != nil {
+		go func() {
+			_ = conn.Close() // Close the connection if writer creation fails
+		}()
+	}
```

do you want to return an error here as well?

---

No.  An error on Close() on something we didn't write to doesn't really have any effect and is generally an OS level issue or broken connection.  But we haven't actually written anything so it isn't important.

---

So I'm taking these back.  I looked at these again and decided to actually use my common packages for this.

So when the writer (lowest level) attempts to close a conn, it spins out a goroutine using a pool.  I track the number of these in case the close blocks forever (the way conns were blocking because of bad domain socket moutning in the RP).  So the upper levels can refuse to create new connections if 10 lower level conns block.  

Nothing should freeze with this.